diff -up ./src/fuser.c.ori ./src/fuser.c
--- ./src/fuser.c.ori	2017-03-25 00:17:53.169339912 +0100
+++ ./src/fuser.c	2019-08-23 11:58:55.268835489 +0200
@@ -187,6 +187,9 @@ scan_procs(struct names *names_head, str
 	pid_t pid, my_pid;
 	uid_t uid;
 
+	if ( (ino_head == NULL) && (dev_head == NULL) )
+		return;
+
 	if ((topproc_dir = opendir("/proc")) == NULL) {
 		fprintf(stderr, _("Cannot open /proc directory: %s\n"),
 			strerror(errno));
@@ -1860,6 +1863,10 @@ scan_knfsd(struct names *names_head, str
 	char *find_space;
 	struct stat st;
 
+	if ( (ino_head == NULL) && (dev_head == NULL) )
+		return;
+
+
 	if ((fp = fopen(KNFSD_EXPORTS, "r")) == NULL) {
 #ifdef DEBUG
 		printf("Cannot open %s\n", KNFSD_EXPORTS);
@@ -1906,6 +1913,10 @@ scan_mounts(struct names *names_head, st
 	char *find_space;
 	struct stat st;
 
+	if ( (ino_head == NULL) && (dev_head == NULL) )
+		return;
+
+
 	if ((fp = fopen(PROC_MOUNTS, "r")) == NULL) {
 		fprintf(stderr, "Cannot open %s\n", PROC_MOUNTS);
 		return;
@@ -1949,6 +1960,9 @@ scan_swaps(struct names *names_head, str
 	char *find_space;
 	struct stat st;
 
+	if ( (ino_head == NULL) && (dev_head == NULL) )
+		return;
+
 	if ((fp = fopen(PROC_SWAPS, "r")) == NULL) {
 		/*fprintf(stderr, "Cannot open %s\n", PROC_SWAPS); */
 		return;
